pkg/image/airgap.go

```diff
@@ -675,6 +702,147 @@ func reportWriterWithProgress(imageInfos map[string]*imagetypes.ImageInfo, repor
 	return pipeWriter
 }
 
+func PushEmbeddedClusterArtifacts(airgapBundle string, opts imagetypes.PushEmbeddedClusterArtifactsOptions) error {
+	tmpDir, err := os.MkdirTemp("", "embedded-cluster-artifacts")
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp directory")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	fileReader, err := os.Open(airgapBundle)
+	if err != nil {
+		return errors.Wrap(err, "failed to open file")
+	}
+	defer fileReader.Close()
+
+	gzipReader, err := gzip.NewReader(fileReader)
+	if err != nil {
+		return errors.Wrap(err, "failed to get new gzip reader")
+	}
+	defer gzipReader.Close()
+
+	tarReader := tar.NewReader(gzipReader)
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.Wrap(err, "failed to get read archive")
+		}
+
+		if header.Typeflag == tar.TypeDir {
+			continue
+		}
+
+		if filepath.Dir(header.Name) != "embedded-cluster" {
+			continue
+		}
+
+		if err := func() error {
+			dstFilePath := filepath.Join(tmpDir, header.Name)
+			if err := os.MkdirAll(filepath.Dir(dstFilePath), 0755); err != nil {
+				return errors.Wrap(err, "failed to create path")
+			}
+			defer os.RemoveAll(dstFilePath)
```

Comment:
this might remove the original tmp directory created at the top if there's no subdirectory, but i guess the above `MkdirAll` will recreate it again? not sure if that will have unexpected behaviors.